src/meta/src/hummock/compaction/base_level_compaction_picker.rs

```diff
@@ -110,6 +110,9 @@ impl CompactionPicker for LevelCompactionPicker {
                 if level_handlers[0].is_level_pending_compact(level) {
                     break;
                 }
+                if level.level_type() != LevelType::Nonoverlapping {
```

Comment:
IMO, we dont need to break when meet `Overlapping` sub_level, because we choose the whole l0 to compact.

So we can ensure the level_type of sub_level by using `level_type: sub_level[level].level_type`

Reply:
The only problem of "not break when meet overlapping sub level" is, it breaks the assumption that all sub levels should go from overlapping to non-overlapping, before being compacted to lower levels.

I think it's OK and +1 for Li's suggestions. @Little-Wallace @hzxa21

Reply:
No. The overlapping level may be small in usually but the size of this compact task may be large because it include the files in base level. So this compact task may pending a long time and it would make the overlapping level exist for a long time. If we compact the overlapping level in an independent task we can make it finish as soon as possible.